frontend/components/charities/CampaignCharityList.tsx

```diff
@@ -0,0 +1,31 @@
+import { Box, Stack } from '@mui/material';
+import React from 'react';
+import { CampaignCharityDonationPublicData } from '../../types/campaignCharities';
+import CampaignCharityListItem from './CampaignCharityListItem';
+
+type Props = {
+  campaignCharities: CampaignCharityDonationPublicData[];
+};
+
+export default function CampaignCharityList({ campaignCharities }: Props) {
+  const maxTotalDonation = Math.max(
+    ...campaignCharities.map(
+      (campaignCharity) => campaignCharity.primaryDonor.amount + campaignCharity.secondaryDonors.amount,
+    ),
+  );
+
+  const getFractionOfMaxDonation = (campaignCharity: CampaignCharityDonationPublicData) => {
+    const total = campaignCharity.primaryDonor.amount + campaignCharity.secondaryDonors.amount;
+    return total / maxTotalDonation;
+  };
+
+  return (
+    <Stack spacing={2}>
+      {campaignCharities.map((campaignCharity, index) => (
+        <Box key={index} style={{ width: `calc(80% * ${getFractionOfMaxDonation(campaignCharity)})` }}>
```

Comment:
<img width="1337" alt="Screenshot 2022-10-16 at 1 06 13 PM" src="https://user-images.githubusercontent.com/71375383/196019144-35703a05-6ea3-44e8-af60-5d8e25a48cde.png">

Please fix, this is infinitely long when there are no donations

Reply:
#168

Reply:
<img width="1214" alt="image" src="https://user-images.githubusercontent.com/75519136/196025661-104bf368-508e-4915-8a73-8b39c6be21a2.png">

Reply:
We need the name of the charity too.

Reply:
![image](https://user-images.githubusercontent.com/53614454/196026304-57632ca3-329a-4ea9-9f44-e681f84bc1c8.png)

Something like this but without the givingSg link.